From 6b67d41736fc998eba8c98a35ee712d0d442611d Mon Sep 17 00:00:00 2001
From: Radu-Andrei BULIE <radu.bulie@freescale.com>
Date: Thu, 16 Aug 2012 23:38:47 +0000
Subject: [PATCH 285/518] dpa_ipsec: Fix incorrect value for protocol table
 selection

Fixed incorrect default protocol table selection

Signed-off-by: Radu-Andrei BULIE <radu.bulie@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_ipsec.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
index 0580377..ff69e1c 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
@@ -207,9 +207,9 @@ static inline void store_ipsec_params(struct dpa_ipsec *dpa_ipsec,
 							DPA_OFFLD_DESC_NONE) {
 			/* IPV4 table desc are at even indexes (IPV6 at odd) */
 			if (i & 0x01)
-				pre_sec_out_params->table[i] = *any_ipv4_table;
-			else
 				pre_sec_out_params->table[i] = *any_ipv6_table;
+			else
+				pre_sec_out_params->table[i] = *any_ipv4_table;
 		}
 	}
 }
-- 
1.7.5.4

